libmscore/measure.cpp

```diff
@@ -3883,22 +3883,23 @@ void Measure::checkTrailer()
 
 void Measure::setStretchedWidth(qreal w)
       {
-      qreal minWidth = score()->styleP(Sid::minMeasureWidth);
+      qreal minWidth = isMMRest() ? score()->styleP(Sid::minMMRestWidth) : score()->styleP(Sid::minMeasureWidth);
       if (w < minWidth)
             w = minWidth;
 
-      // multi measure rests are not stretched depending on their
-      // tick length
-
-      if (!isMMRest()) {
-            qreal stretchableWidth = 0.0;
-            for (Segment* s = first(SegmentType::ChordRest); s; s = s->next(SegmentType::ChordRest)) {
-                  if (!s->enabled())
-                        continue;
-                  stretchableWidth += s->width();
-                  }
-            w += stretchableWidth * (basicStretch()-1.0) * ticks().ticks() / 1920.0;
+      qreal stretchableWidth = 0.0;
+      for (Segment* s = first(SegmentType::ChordRest); s; s = s->next(SegmentType::ChordRest)) {
+            if (!s->enabled())
+                  continue;
+            stretchableWidth += s->width();
             }
+      int weight = ticks().ticks();
+      // reduce weight of mmrests
+      // so the nominal width is not directly proportional to duration
+      if (isMMRest())
+            weight = (weight >> 2) + 1;
```

Comment:
Also I see that the weight calculation logic is commonly duplicated within this PR. Maybe it can be extracted to a separate `Measure` class member function, something like
```
int Measure::layoutWeight() const
      {
      int weight = ticks().ticks();
      if (isMMRest())
            weight = (weight >> 2) + 1;
      return weight;
      }
```

Also, are there any reasons to write exactly `weight >> 2` instead of `weight / 4` or something similar?

Reply:
First: I have this notion from way back when that >> 2 is faster, probably most compilers are smart enough for this to not be true though.  I see the >> 2 used in track / voice calculations elsewhere, probably @wschweer comes from the same mindset :-)

As for the refactor, I did consider that.  One thing that I didn't make clear: while the two calculations in layout.cpp need to be in sync, the one in measure.cpp could actually be different and it would still work.  The measure.cpp calculation is about calculating a default width for the measure, the layout.cpp calculation is about determining what proportion of the "rest" of the system should be allocated to the measure during system fill/stretch.  But I didn't see any reason not to use the same calculation.

So, I'll go ahead and do the refactor but add a comment.